src/behavior.rs

```diff
@@ -59,50 +85,109 @@ pub trait Behavior<Pane> {
+                galley.size().x + (2.0 * x_margin),
+                ui.available_height(),
+            ))
+        };
+
+        let tab_response = ui.interact(tab_rect, id, Sense::click_and_drag());
 
         // Show a gap when dragged
-        if ui.is_rect_visible(rect) && !is_being_dragged {
-            let bg_color = self.tab_bg_color(ui.visuals(), tiles, tile_id, active);
-            let stroke = self.tab_outline_stroke(ui.visuals(), tiles, tile_id, active);
-            ui.painter().rect(rect.shrink(0.5), 0.0, bg_color, stroke);
+        if ui.is_rect_visible(tab_rect) && !state.is_being_dragged {
+            let bg_color = self.tab_bg_color(ui.visuals(), tiles, tile_id, &state);
+            let stroke = self.tab_outline_stroke(ui.visuals(), tiles, tile_id, &state);
+            ui.painter()
+                .rect(tab_rect.shrink(0.5), 0.0, bg_color, stroke);
 
-            if active {
+            if state.active {
                 // Make the tab name area connect with the tab ui area:
                 ui.painter().hline(
-                    rect.x_range(),
-                    rect.bottom(),
+                    tab_rect.x_range(),
+                    tab_rect.bottom(),
                     Stroke::new(stroke.width + 1.0, bg_color),
                 );
             }
 
-            let text_color = self.tab_text_color(ui.visuals(), tiles, tile_id, active);
-            ui.painter().galley(
-                egui::Align2::CENTER_CENTER
-                    .align_size_within_rect(galley.size(), rect)
-                    .min,
-                galley,
-                text_color,
-            );
+            // Prepare title's text for rendering
+            let text_color = self.tab_text_color(ui.visuals(), tiles, tile_id, state.active);
+            let mut text_position = egui::Align2::LEFT_CENTER
+                .align_size_within_rect(galley.size(), tab_rect)
+                .min;
+
+            // Respect the left margin
+            text_position.x += x_margin;
+
+            // Render the title
+            ui.painter().galley(text_position, galley, text_color);
+
+            // Conditionally render the close button
+            if state.closable {
+                let close_btn_pad = (tab_rect.height() - close_btn_size.y) / 2.0;
+                let close_btn_rect = Rect::from_min_size(
+                    Pos2::from([
+                        tab_rect.right() - close_btn_pad - close_btn_size.x,
+                        tab_rect.center().y - 0.5 * close_btn_size.y,
+                    ]),
+                    close_btn_size,
+                );
+
+                // Allocate
+                let close_btn_id = ui.auto_id_with("tab_close_btn");
+                let close_btn_response = ui.interact(close_btn_rect, close_btn_id, Sense::click());
+
+                // Update the current UI's rect to include the close button
+                ui.expand_to_include_rect(close_btn_response.rect);
```

Comment:
This shouldn't be needed - `ui.allocate_space` has already allocated the superset `tab_rect`